src/django_components/component.py

```diff
@@ -2692,13 +2796,38 @@ def _render_impl(
         if not isinstance(context, Context):
             context = RequestContext(request, context) if request else Context(context)
 
-        render_id = COMP_ID_PREFIX + gen_id()
+        render_id = _gen_component_id()
+
+        # Apply defaults to missing or `None` values in `kwargs`
+        defaults = defaults_by_component.get(comp_cls, None)
+        if defaults is not None:
+            apply_defaults(kwargs_dict, defaults)
+
+        # If user doesn't specify `Args`, `Kwargs`, `Slots` types, then we pass them in as plain
+        # dicts / lists.
+        args_inst = comp_cls.Args(*args_list) if comp_cls.Args is not None else args_list
```

Comment:
This was previously on the [line 2782](https://github.com/django-components/django-components/pull/1212/files#diff-35c84ebceae50f7149f03bbcaf2b4f1289bc2984bf7859ca521b5dbe7960c396L2782)